compiler/x/amd64/codegen/OMRMemoryReference.cpp

```diff
@@ -294,6 +294,8 @@ bool OMR::X86::AMD64::MemoryReference::needsAddressLoadInstruction(intptr_t next
       return true;
    else if (sr.getSymbol() && (sr.getSymbol()->isBlockFrequency() || sr.getSymbol()->isRecompQueuedFlag()) && cg->needRelocationsForPersistentProfileInfoData())
       return true;
+   else if (sr.getSymbol() && sr.getSymbol()->isCatchBlockCounter())
```

Comment:
Don't we need to qualify this by something similar to (if not the same as) `needRelocationsForBodyInfoData()`?
This query returns false for a normal JIT compilation, but true for AOT or remote compilations.

Reply:
It does. Fixed.